scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+name: CreateExp
+decl: static
+details:
+    - "Metric handles from multiple domains may be used for a single metric group."
+    - "Metric handles from different sourceIds (refer $t_metric_programmable_exp_properties_t) are not allowed in a single metric group."
+params:
+    - type: $t_device_handle_t
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: const char
+      name: name[ZET_MAX_METRIC_GROUP_NAME]
+      desc: "[in] Metric group name."
+    - type: const char
+      name: description[ZET_MAX_METRIC_GROUP_DESCRIPTION]
+      desc: "[in] Metric group description."
+    - type: $t_metric_group_handle_t*
+      name: phMetricGroup
+      desc: "[out] Created Metric group handle"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Add a metric handle to the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: AddMetricExp
+decl: static
+details:
+    - "Reasons for failing to add the metric could be queried using pErrorString"
+params:
+    - type: $t_metric_group_handle_t 
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric to be added to the group."
+    - type: "uint32_t*"
+      name: pErrorStringSize
+      desc: "[in,out][optional] Size of the error string to query, if an error was reported during adding the metric handle."
+    - type: "char*"
+      name: pErrorString
+      desc: |
+            [in,out][optional][range(0, *pErrorStringSize)] Error string.
+            if *pErrorStringSize is less than the length of the error string available, then driver shall only retrieve that length of error string.
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If a Metric handle from a pre-defined metric group is requested to be added"
+    - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
+        - "If the Metric group handle is already activated"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Remove a metric from the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: RemoveMetricExp
+decl: static
+details:
+    - "Remove an already added metric handle from the metric group"
+params:
+    - type: $t_metric_group_handle_t
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric handle to be removed from the metric group."
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If trying to remove a metric not previously added to the metric group"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Finalizes a created metric group using $tMetricGroupCreateExp, so that it can be activated."
+version: "1.8"
+class: $tMetricGroup
+name: FinalizeExp
```

Comment:
if the counter API of this is tMetricGroupCreateExp, then this should be called Destroy, following the convention of other APIs in the spec.

Reply:
but I see there's a destroy below. what is finalize and what is destroy? why do we need both? it seems confusing as both refer to tMetricGroupCreateExp

>"Finalizes a created metric group using $tMetricGroupCreateExp
> "Destroy a metric group created using $tMetricGroupCreateExp."

Create goes with Destroy. Finalize seems to need a Start or Begin interface.

Reply:
>if the counter API of this is tMetricGroupCreateExp,
This API is NOT a counter API for MetricGroupCreate.
This API is for finalizing (stop configuring by using the ADD / REMOVE APIs) the metric group.

Just a note: We have MetricGroupDestroy as well.
Thanks

Reply:
> it seems confusing as both refer to tMetricGroupCreateExp

Ok . Let me update the description to be more precise about its usage.

Reply:
thanks @joshuaranjan . I still dont understand what Finalize does. could you share pseudo code  or add a section in [scripts/tools/EXT_Exp_MetricProgrammable.rst](https://github.com/oneapi-src/level-zero-spec/pull/204/files#diff-04edfdabb9ef36b6fc56e99cd7e7cb4a1df5867167c10f78cec8b6aec3b1570f) showing how it is used?